interface/wx/regex.h

```diff
@@ -250,5 +250,35 @@ class wxRegEx
         Replace the first occurrence.
     */
     int ReplaceFirst(wxString* text, const wxString& replacement) const;
+
+    /**
+        Escapes any of the following \, ^, $, ., |, ?, *, +, (, ), [, ], { and } metacharacters
+        found in the passed string with a backslash.
+
+        @param str
+            A string that may contain metacharacters to escape.
+
+        @return A string with all metacharacters converted to their escaped form.
+
+        @note This is a static function, and there is no need to create an instance of wxRegEx to use it.
```

Comment:
I don't think this is necessary, we assume basic knowledge of C++ and don't add such notes to the other static methods.

OTOH `@see Unescape` could be useful to add here.

And maybe also an example showing some simple input and output.